src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx

```diff
@@ -35,6 +40,24 @@ const HotspotSetupWifiConnectingScreen = () => {
 
   const { showOKAlert } = useAlert()
 
+  const goToNextStep = async () => {
+    if (connectedHotspot.address) {
+      const address = await getAddress()
+      const hotspot = await getHotspotDetails(connectedHotspot.address)
+      if (hotspot && hotspot.owner === address) {
+        navigation.replace('OwnedHotspotErrorScreen')
+      } else if (hotspot && hotspot.owner !== address) {
+        navigation.replace('NotHotspotOwnerErrorScreen')
+      } else {
+        navigation.replace('HotspotSetupLocationInfoScreen')
+      }
+    } else {
+      Logger.error('no connectedHotspot address after connecting to wifi')
+      showOKAlert({ titleKey: 'something went wrong' })
```

Comment:
should we show a more informative error?

Reply:
Made it `There was an error connecting to the Hotspot. Please try again.`